src/test/java/org/gridsuite/modification/server/service/BuildTest.java

```diff
@@ -349,6 +349,123 @@ public void runBuildWithEmptyGroupTest() throws Exception {
+            TEST_REPORT_ID,
+            List.of(TEST_GROUP_ID),
+            List.of(TEST_SUB_REPORTER_ID_1),
+            new HashSet<>());
+
+        // Build variant
+        networkModificationService.buildVariant(TEST_NETWORK_ID, buildInfos);
+
+        // Check if added equipments are indexed in elasticsearch
+        //check voltage level indexation
+        var expectedEquipmentInfos = EquipmentInfos.builder()
+                .networkUuid(TEST_NETWORK_ID)
+                .variantId(NetworkCreation.VARIANT_ID)
+                .id("vl1")
+                .name("vl1")
+                .type(IdentifiableType.VOLTAGE_LEVEL.name())
+                .voltageLevels(Set.of(VoltageLevelInfos.builder().id("vl1").name("vl1").build()))
+                .substations(Set.of(SubstationInfos.builder().id("s1").name("s1").build()))
+                .build();
+        var equipmentInfos = equipmentInfosRepository.findByIdInAndNetworkUuidAndVariantId(List.of("vl1"), TEST_NETWORK_ID, NetworkCreation.VARIANT_ID).get(0);
+        assertTrue(areEquipmentInfosEqual(expectedEquipmentInfos, equipmentInfos));
+
+        //check load indexation
+        expectedEquipmentInfos = EquipmentInfos.builder()
+                .networkUuid(TEST_NETWORK_ID)
+                .variantId(NetworkCreation.VARIANT_ID)
+                .id("newLoad")
+                .name("newLoad")
+                .type(IdentifiableType.LOAD.name())
+                .voltageLevels(Set.of(VoltageLevelInfos.builder().id("vl1").name("vl1").build()))
+                .substations(Set.of(SubstationInfos.builder().id("s1").name("s1").build()))
+                .build();
+        equipmentInfos = equipmentInfosRepository.findByIdInAndNetworkUuidAndVariantId(List.of("newLoad"), TEST_NETWORK_ID, NetworkCreation.VARIANT_ID).get(0);
+        assertTrue(areEquipmentInfosEqual(expectedEquipmentInfos, equipmentInfos));
+
+        //check line indexation
+        expectedEquipmentInfos = EquipmentInfos.builder()
+                .networkUuid(TEST_NETWORK_ID)
+                .variantId(NetworkCreation.VARIANT_ID)
+                .id("newLine")
+                .name("newLine")
+                .type(IdentifiableType.LINE.name())
+                .voltageLevels(Set.of(VoltageLevelInfos.builder().id("v1").name("v1").build(), VoltageLevelInfos.builder().id("v2").name("v2").build()))
+                .substations(Set.of(SubstationInfos.builder().id("s1").name("s1").build()))
+                .build();
+        equipmentInfos = equipmentInfosRepository.findByIdInAndNetworkUuidAndVariantId(List.of("newLine"), TEST_NETWORK_ID, NetworkCreation.VARIANT_ID).get(0);
+        assertTrue(areEquipmentInfosEqual(expectedEquipmentInfos, equipmentInfos));
+
+        assertTrue(TestUtils.getRequestsDone(1, server).stream().anyMatch(r -> r.matches("/v1/reports/.*")));
+    }
+
+    public static boolean areEquipmentInfosEqual(EquipmentInfos equipmentInfos1, EquipmentInfos equipmentInfos2) {
```

Comment:
Remove this method and use an utility method for  equality test
Or use directly the `equals` method of `EquipmentInfo`